libs/agno/agno/document/reader/arxiv_reader.py

```diff
@@ -39,3 +40,14 @@ def read(self, query: str) -> List[Document]:
             )
 
         return documents
+
+    async def async_read(self, query: str) -> List[Document]:
+        """
+        Search a query from arXiv database asynchronously
+
+        This function gets the top_k articles based on a user's query, sorted by relevance from arxiv
+
+        @param query:
+        @return: List of documents
+        """
+        return await asyncio.to_thread(self.read, query)
```

Comment:
I don't think we want to do this. This means it will go off concurrently to the main loop. If you wanted to use a thread, you'll have to take the task and wait for its response before returning. Because otherwise this response would always be empty?

Reply:
But somehow it does work...

Reply:
nvm, I looked it up, we're good.

Reply:
you mean like this-
```py
task = asyncio.create_task(asyncio.to_thread(self.read, query))
        result = await task 
        return result
```

but we are already awaiting the thread's response, shouldn't that be enough? If we weren't awaiting then it could cause problems. But ig creating a task might be more robust.

Reply:
Yeah your right, all good

Reply:
but i did the change, should i revert to `to_thread()`